src/java/org/apache/cassandra/schema/CompressionParams.java

```
@@ -20,184 +20,355 @@
+    private final ICompressor sstableCompressor;
+    /**
+     * The chunk length in KB
+     */
+    private final int chunkLength;
+    /**
+     * The compressed length in KB.
+     * In content we store max length to avoid rounding errors causing compress/decompress mismatch.
+     */
+    private final int maxCompressedLength;
+    /**
+     * The minimum compression ratio.
+     * In configuration, we store min ratio, the input parameter.
+     * Ths is mathematically related to chunkLength and maxCompressedLength in that
+     * # chunk_length / max_compressed_length = min_compress_ratio
+     */
+    private final double minCompressRatio;
+    /**
+     * unrecognized options, may contain compressor specific options
+     */
+    private final ImmutableMap<String, String> otherOptions;
 
-        if (!removeEnabled(options) && !options.isEmpty())
-            throw new ConfigurationException(format("If the '%s' option is set to false no other options must be specified", ENABLED));
-        else
-            sstableCompressionClass = removeSSTableCompressionClass(options);
+    /**
+     * Gets the default compression params for the keyspace.
+     * This method accounts for issues when the compression params change for system keyspaces.
+     *
+     * @param keyspace the name of the keyspace to get params for. (may be null for non-System keyspaces)
+     * @return The compresson parameters for the keyspace.
+     */
+    public static CompressionParams defaultParams(String keyspace)
+    {
+        if (keyspace != null && SchemaConstants.getLocalAndReplicatedSystemKeyspaceNames().contains(keyspace))
+            return DEFAULT;
 
-        int chunkLength = removeChunkLength(options);
-        double minCompressRatio = removeMinCompressRatio(options);
+        CompressionParams result = CALCULATED_DEFAULT;
+        if (result == null)
+        {
+            if (DatabaseDescriptor.getRawConfig() == null)
```

Should we throw an exception in this case if the config is not initialized when this method is called?